purchase_request_group_by_date/models/stock_rule.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2018-2020 ForgeFlow, S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+from datetime import datetime
+
+from odoo import api, fields, models
+
+
+class StockRule(models.Model):
+    _inherit = "stock.rule"
+
+    company_group_purchase_request = fields.Boolean(
+        related="company_id.group_purchase_request"
+    )
+
+    # override to change returned domain based on company settings
+    @api.model
+    def _make_pr_get_domain(self, values):
+        domain = super()._make_pr_get_domain(values)
+        if self.company_group_purchase_request:
+            domain = (
+                ("picking_type_id", "=", self.picking_type_id.id),
+                ("company_id", "=", values["company_id"].id),
+                ("date_start", "=", datetime.today().date()),
+            )
+            gpo = self.group_propagation_option
+            group_id = (
+                (gpo == "fixed" and self.group_id.id)
+                or (gpo == "propagate" and values["group_id"].id)
+                or False
+            )
+            if group_id:
+                domain += (("group_id", "=", group_id),)
+        return domain
+
+    # override to change existing pr handling
+    def create_purchase_request(self, procurement_group):
```

Comment:
It's quite hard to find out what has changed in this method.
Why not splitting the method in `purchase_request` and override just what you need?
Would be easier to understand, I think.

Reply:
i'd prefer to not go too deep with modification, esp. in original module.
in case of _make_pr_get_domain i think due to method construction is it clear what we want to change. 
things get bit worse in create_purchase_request method but crucial part of method is imo well commented.